README.md

```diff
@@ -2,70 +2,237 @@
+        "index": "3",
+        "initrd": "/boot/initramfs-0-rescue-c44543d15b2c4e898912c2497f734e67.img",
+        "kernel": "/boot/vmlinuz-0-rescue-c44543d15b2c4e898912c2497f734e67",
+        "root": "UUID=65c70529-e9ad-4778-9001-18fe8c525285",
+        "title": "Fedora Linux (0-rescue-c44543d15b2c4e898912c2497f734e67) 36 (Workstation Edition)"
+    },
+    {
+        "args": "ro rootflags=subvol=root rd.luks.uuid=luks-9da1fdf5-14ac-49fd-a388-8b1ee48f3df1 rhgb quiet $tuned_params",
+        "id": "luks-9da1fdf5-14ac-49fd-a388-8b1ee48f3df1 rhgb quiet $tuned_params",
+        "index": "2",
+        "initrd": "/boot/initramfs-6.3.12-100.fc37.x86_64.img $tuned_initrd",
+        "kernel": "/boot/vmlinuz-6.3.12-100.fc37.x86_64",
+        "root": "UUID=65c70529-e9ad-4778-9001-18fe8c525285",
+        "title": "Fedora Linux (6.3.12-100.fc37.x86_64) 37 (Workstation Edition)"
+    },
+    {
+        "args": "ro rootflags=subvol=root rd.luks.uuid=luks-9da1fdf5-14ac-49fd-a388-8b1ee48f3df1 rhgb quiet $tuned_params",
+        "id": "luks-9da1fdf5-14ac-49fd-a388-8b1ee48f3df1 rhgb quiet $tuned_params",
+        "index": "1",
+        "initrd": "/boot/initramfs-6.4.15-100.fc37.x86_64.img $tuned_initrd",
+        "kernel": "/boot/vmlinuz-6.4.15-100.fc37.x86_64",
+        "root": "UUID=65c70529-e9ad-4778-9001-18fe8c525285",
+        "title": "Fedora Linux (6.4.15-100.fc37.x86_64) 37 (Workstation Edition)"
+    },
+    {
+        "args": "ro rootflags=subvol=root rd.luks.uuid=luks-9da1fdf5-14ac-49fd-a388-8b1ee48f3df1 rhgb quiet $tuned_params",
+        "id": "luks-9da1fdf5-14ac-49fd-a388-8b1ee48f3df1 rhgb quiet $tuned_params",
+        "index": "0",
+        "initrd": "/boot/initramfs-6.5.7-100.fc37.x86_64.img $tuned_initrd",
+        "kernel": "/boot/vmlinuz-6.5.7-100.fc37.x86_64",
+        "root": "UUID=65c70529-e9ad-4778-9001-18fe8c525285",
+        "title": "Fedora Linux (6.5.7-100.fc37.x86_64) 37 (Workstation Edition)"
+    }
+]
+```
+
+## Example Playbook
 
 ```yaml
 - hosts: all
   vars:
-    bootloader_foo: "foo foo!"
-    bootloader_bar: "progress bar"
-
+    bootloader_settings:
+      - kernel:
+          kernel_path: /boot/vmlinuz-0-rescue-1
+        options:
+          - name: console
+            value: tty0
+            state: present
+          - previous: replaced
+      - kernel:
+          kernel_index: [1, 2, 3]
+        options:
+          - name: print-fatal-signals
+            value: 1
+      - kernel:
+          kernel_title: Red Hat Enterprise Linux (4.1.1.1.el8.x86_64) 8
+        options:
+          - name: no_timer_check
+      - kernel:
+          kernel_path: /boot/vmlinuz-0-rescue-1
+        options:
+          - name: console
+            value: tty0
+          - name: print-fatal-signals
+            value: 1
+          - name: no_timer_check
+            state: present
+          - previous: replaced
+      - kernel: ALL
+        options:
+          - name: debug
+            state: present
+      - kernel: DEFAULT
```

Comment:
I guess this is the same as `index: 0`

Reply:
It is, from `man grubby`; 
```
       There are a number of ways to specify the kernel used for --info, --remove-kernel, and --update-kernel. Specificying DEFAULT or ALL selects the default entry and
       all of the entries, respectively.  Also, the title of a boot entry may be specified by using TITLE=title as the argument; all entries with that title are used.
```

ALL and DEFAULT are the only supported strings, so the role should know how to work with them.